src/convertToObject.js

```diff
@@ -12,7 +12,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const allProperties = sourceString
+    .split(';')
+    .map(property => property.trim())
+    .filter(property => property.length > 0);
+
+  const keys = allProperties.map(property => property.split(':')[0].trim());
+  const values = allProperties.map(property => property.split(':')[1].trim());
+
+  const result = {};
+
+  for (let i = 0; i < keys.length; i++) {
+    result[keys[i]] = values[i];
+  }
```

Comment:
Try to avoid loops. Use array methods, for example, `forEach()`.

Reply:
thanks, I fixed it